client/java/src/main/java/io/openlineage/client/utils/jdbc/OracleJdbcExtractor.java

```diff
@@ -39,6 +39,59 @@ public JdbcLocation extract(String rawUri, Properties properties) throws URISynt
   }
 
   private JdbcLocation extractUri(String uri, Properties properties) throws URISyntaxException {
+    // Handle LDAP connection strings, including cases with multiple LDAP URIs
+    if (uri.toLowerCase(Locale.ROOT).contains("ldap")) {
+      // Split original URI by whitespace to cover scenarios with multiple LDAP URIs
+      String[] tokens = uri.split("\\\\s+");
+      String firstLdapUri = null;
+      for (String token : tokens) {
+        String lowerToken = token.toLowerCase(Locale.ROOT);
+        if (lowerToken.contains("ldap://") || lowerToken.contains("ldaps://") ||
+                lowerToken.contains("ldap:") || lowerToken.contains("ldaps:")) {
+          // In case a prefix like "jdbc:oracle:thin:@" exists, take the substring starting from "ldap"
+          int idx = lowerToken.indexOf("ldap");
+          firstLdapUri = token.substring(idx);
+          break;
+        }
+      }
+
+      if (firstLdapUri != null) {
+        String lowerUri = firstLdapUri.toLowerCase(Locale.ROOT);
+        String prefix = "";
+        if (lowerUri.startsWith("ldap://")) {
+          prefix = "ldap://";
+        } else if (lowerUri.startsWith("ldaps://")) {
+          prefix = "ldaps://";
+        } else if (lowerUri.startsWith("ldap:")) {
+          prefix = "ldap:";
+        } else if (lowerUri.startsWith("ldaps:")) {
+          prefix = "ldaps:";
+        }
+
+        if (!prefix.isEmpty()) {
+          // Remove the LDAP prefix and any extra leading slashes
+          String withoutPrefix = firstLdapUri.substring(prefix.length());
+          while (withoutPrefix.startsWith("/")) {
+            withoutPrefix = withoutPrefix.substring(1);
+          }
```

Comment:
Probably all these comparisons could be replaced with regexp match group `ldaps?:/{0,2}(.+)`